ckan/logic/action/create.py

```diff
@@ -1059,24 +1058,16 @@ def user_invite(context: Context,
         raise NotFound()
 
     name = _get_random_username_from_email(data['email'])
-    # Choose a password. However it will not be used - the invitee will not be
-    # told it - they will need to reset it
-    while True:
-        password = ''.join(random.SystemRandom().choice(
-            string.ascii_lowercase + string.ascii_uppercase + string.digits)
-            for _ in range(12))
-        # Occasionally it won't meet the constraints, so check
-        validation_errors: FlattenErrorDict = {}
-        ckan.logic.validators.user_password_validator(
-            ('password', ), {('password', ): password},
-            validation_errors, context)
-        if not validation_errors:
-            break
 
     data['name'] = name
-    data['password'] = password
+    # send the proper schema when creating a user from here
+    # so the password field would be ignored.
+    context['schema'] = ckan.logic.schema.create_user_for_user_invite_schema()
     data['state'] = model.State.PENDING
     user_dict = _get_action('user_create')(context, data)
+    # we already created the user so we don't need the schema anymore
+    # as we are calling for other actions bellow
+    context.pop('schema')
```

Comment:
```suggestion
    invite_schema = ckan.logic.schema.create_user_for_user_invite_schema()
    data['state'] = model.State.PENDING
    user_dict = _get_action('user_create')(dict(context, schema=invite_schema), data)
    # we already created the user so we don't need the schema anymore
    # as we are calling for other actions bellow
```
passing in new dicts to context is safer and means we don't have to remember to `pop` anything

Reply:
Yep, forgot about that :D , thanks